test/parallel_api/ranges/std_ranges_test.h

```diff
@@ -625,61 +630,102 @@ struct test_range_algo
     test_range_algo(std::array<int, 2> sizes) : n_serial(sizes[0]), n_parallel(sizes[1]) {}
 #endif
 
-    void test_view(auto view, auto algo, auto& checker, auto... args)
+    void test_view_host(auto view, auto algo, auto& checker, auto... args)
     {
         test<T, host_subrange<T>, mode, DataGen1, DataGen2>{}.host_policies(n_serial, n_parallel, algo, checker, view, std::identity{}, args...);
+    }
+
 #if TEST_DPCPP_BACKEND_PRESENT
-        test<T, usm_subrange<T>, mode, DataGen1, DataGen2>{}(n_device, TestUtils::get_dpcpp_test_policy<call_id>(), algo, checker, view, std::identity{}, args...);
-#endif //TEST_DPCPP_BACKEND_PRESENT
+    template <typename Policy>
+    void test_view_hetero(Policy&& exec, auto view, auto algo, auto& checker, auto... args)
+    {
+        test<T, usm_subrange<T>, mode, DataGen1, DataGen2>{}(n_device, CLONE_TEST_POLICY_IDX(exec, call_id), algo, checker, view, std::identity{}, args...);
     }
+#endif //TEST_DPCPP_BACKEND_PRESENT
+
+    struct subrange_view_fo
+    {
+        template <typename TArg>
+        auto operator()(TArg&& v) const
+        {
+            return std::ranges::subrange(v);
+        }
+    };
 
-    void operator()(auto algo, auto& checker, auto... args)
+#if TEST_CPP20_SPAN_PRESENT
+    struct span_view_fo
     {
+        template <typename TArg>
+        auto operator()(TArg&& v) const
+        {
+            return std::span(v);
+        }
+    };
+#endif
 
-        auto subrange_view = [](auto&& v) { return std::ranges::subrange(v); };
+    void
+    test_range_algo_impl_host(auto algo, auto& checker, auto... args)
+    {
+        auto subrange_view = subrange_view_fo{};
 #if TEST_CPP20_SPAN_PRESENT
-        auto span_view = [](auto&& v) { return std::span(v); };
+        auto span_view = span_view_fo{};
 #endif
 
-        test<T, host_vector<T>, mode, DataGen1, DataGen2>{}.host_policies(
-            n_serial, n_parallel, algo, checker, std::identity{}, std::identity{}, args...);
-        test<T, host_vector<T>, mode, DataGen1, DataGen2>{}.host_policies(
-            n_serial, n_parallel, algo, checker, subrange_view, std::identity{}, args...);
-        test<T, host_vector<T>, mode, DataGen1, DataGen2>{}.host_policies(
-            n_serial, n_parallel, algo, checker, std::views::all, std::identity{}, args...);
-        test<T, host_subrange<T>, mode, DataGen1, DataGen2>{}.host_policies(
-            n_serial, n_parallel, algo, checker, std::views::all, std::identity{}, args...);
+        test<T, host_vector<T>,   mode, DataGen1, DataGen2>{}.host_policies(n_serial, n_parallel, algo, checker, std::identity{},  std::identity{}, args...);
```

Comment:
The lines have become very long 150-180 symbols. It is difficult to read...

Reply:
It's too much for some editors, e.g. vscode:

<img width="2509" height="1331" alt="image" src="https://github.com/user-attachments/assets/7081f785-7753-47ff-8687-3a66c9bf6643" />

Reply:
It's not critical, so I do not insist.